internal/cmd/options.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2022 The OpenVEX Authors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openvex/go-vex/pkg/vex"
+	"github.com/spf13/cobra"
+)
+
+type vexDocOptions struct {
+	DocumentID string
+	Author     string
+	AuthorRole string
+}
+
+const (
+	productLongFlag = "product"
+	vulnLongFlag    = "vuln"
+)
+
+// Validate checks that the document options are valid
+func (do *vexDocOptions) Validate() error {
+	if do.Author == "" {
+		return fmt.Errorf("document author cannot be blank")
+	}
+	return nil
+}
+
+func (do *vexDocOptions) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(
+		&do.DocumentID,
+		"id",
+		"",
+		"ID string for the new VEX document (autogenerated by default)",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.Author,
+		"author",
+		vex.DefaultAuthor,
+		"author to record in the new document",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.AuthorRole,
+		"author-role",
+		vex.DefaultRole,
+		"optional author role to record in the new document",
+	)
+}
+
+type vexStatementOptions struct {
+	Status          string
+	StatusNotes     string
+	Justification   string
+	ImpactStatement string
+	Vulnerability   string
+	ActionStatement string
+	Product         string
+	Subcomponents   []string
+}
+
+// Validate checks that the statement options are coherent
+func (so *vexStatementOptions) Validate() error {
```

Comment:
(this comment is more for a future PR, but just a thought inspired by the discussion below about action statements)

It would probably help to have tests for validate methods like this one. They'd help guide refactoring of the implementation by checking for overall correctness; and, tests would help illustrate to other developers what different kinds of "correct" and "incorrect" data look like.

Reply:
Sounds good to me! I checked an initial pair of tests covering the basic cases of the validation functions implemented at the moment. I'll add the tests for the rest of them as they are improved.